routes/devices_compat.js

```diff
@@ -16,20 +16,26 @@ const EngineManager = require('../almond/enginemanagerclient');
 
 var router = express.Router();
 
-router.use(user.requireLogIn);
-
 router.get('/oauth2/callback/:kind', (req, res, next) => {
-    const kind = req.params.kind;
 
-    EngineManager.get().getEngine(req.user.id).then(async (engine) => {
+    if (!req.session.redirect){
```

Comment:
Please use the same code style as the existing code: 4 spaces indent, and always a space before `{`